### PR TITLE
Export files required by OPT to load the tokenizer

### PR DESCRIPTION
**Pull request by dbogunowicz**
OPT does not contain a "tokenizer.json", but instead requires the following files:
"special_tokens_map.json"
"vocab.json"
"merges.txt"
to create the tokenizer from pretrained.

I.e if those three files are present in the `deployment` model this line of code will execute:
```python
model_path = "deployment"
tokenizer = AutoTokenizer.from_pretrained(
            model_path,
        )
```

### CODE REVIEW

File: src/sparseml/transformers/export.py [hunk 1]
```diff
@@ -97,7 +97,14 @@
     "tokenizer_config.json",
     "config.json",
 ]
-OPTIONAL_DEPLOYMENT_FILES: List[str] = [EXTERNAL_ONNX_DATA_NAME, "tokenizer.json"]
+
+OPTIONAL_DEPLOYMENT_FILES: List[str] = [
+    EXTERNAL_ONNX_DATA_NAME,
+    "tokenizer.json",
+    "special_tokens_map.json",
```
Comment (natuan) [inline src/sparseml/transformers/export.py:104]: to make this self-explains, I would put these 3 new files under a separate list with hint that it's for OPT model, before you do the merge.

**Review by natuan — CHANGES_REQUESTED** [review]

File: src/sparseml/transformers/export.py (hunk 1 above)
Comment (dbogunowicz) [reply src/sparseml/transformers/export.py:104]: Crossed my mind, makes sense.

Comment (bfineran) [reply src/sparseml/transformers/export.py:104]: agree

File: src/sparseml/transformers/export.py [hunk 2]
```diff
@@ -91,13 +91,18 @@
...
     "tokenizer_config.json",
     "config.json",
 ]
-OPTIONAL_DEPLOYMENT_FILES: List[str] = [EXTERNAL_ONNX_DATA_NAME, "tokenizer.json"]
+EXTERNAL_ONNX_DATA_NAME = ["model.data"]
+OPT_TOKENIZER_FILES = ["special_tokens_map.json", "vocab.json", "merges.txt"]
+
+OPTIONAL_DEPLOYMENT_FILES: List[str] = []
+OPTIONAL_DEPLOYMENT_FILES.extend(EXTERNAL_ONNX_DATA_NAME)
+OPTIONAL_DEPLOYMENT_FILES.extend(OPT_TOKENIZER_FILES)
```
Comment (anmarques) [inline src/sparseml/transformers/export.py:104]: tokenizer.json is no longer in the list

File: src/sparseml/transformers/export.py [hunk 3]
```diff
@@ -91,13 +91,18 @@
...
+MANDATORY_DEPLOYMENT_FILES = [
     MODEL_ONNX_NAME,
     "tokenizer_config.json",
     "config.json",
 ]
-OPTIONAL_DEPLOYMENT_FILES: List[str] = [EXTERNAL_ONNX_DATA_NAME, "tokenizer.json"]
+EXTERNAL_ONNX_DATA_NAME = ["model.data"]
+OPT_TOKENIZER_FILES = ["special_tokens_map.json", "vocab.json", "merges.txt"]
+
+OPTIONAL_DEPLOYMENT_FILES: List[str] = []
```
Comment (dbogunowicz) [inline src/sparseml/transformers/export.py:102]:
```suggestion
OPTIONAL_DEPLOYMENT_FILES: List[str] = ["tokenizer.json"]
```

**Review by anmarques — APPROVED** [review]

**Review by natuan — APPROVED** [review]

**Pull request merged**